### PR TITLE
Basically complete the project

### PR DESCRIPTION
**Pull request by captt-g**
Fix: https://github.com/pycasbin/redis-watcher/issues/1

### CODE REVIEW

**Comment by hsluoyz** [comment]
@captt-g 

1. Plz use English.
2. Write good commit message.

File: redis_watcher/main.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+    def set_update_callback(self, fn_name: Callable):
+        print("runtime is set update callback", fn_name)
+        self.update_callback = fn_name
+
+    def update(self):
+        conn = redis.Redis(
+            host=self.host,
+            port=self.port,
+            user=self.db,
```
Comment (YihuiLu) [inline redis_watcher/main.py:85]: This should be a BUG, because Redis does not have a user parameter

**Comment by captt-g** [comment]
@YihuiLu sry, I accidentally read it wrong, already fixed it.

**Comment by hsluoyz** [comment]
@Zxilly plz review.

/cc @techoner

**Comment by techoner** [comment]
@captt-g This is not complete, right? After receiving the message from redis, there is no callback processing

**Review by techoner — CHANGES_REQUESTED** [review]
After receiving the message from redis, do callback processing .

**Comment by captt-g** [comment]
@techoner Yes, now I add the default update_callback() func, you can use it like red_watcher.set_update_callback(red_watcher.update_callback),thanks for your review.

**Comment by techoner** [comment]
@captt-g Please refer to Go and Nodejs:

https://github.com/casbin/redis-watcher
https://github.com/node-casbin/redis-watcher

**Comment by hsluoyz** [comment]
@captt-g hi, who are you? Are you applying for Summer 2021 or just for open source contribution?

**Comment by Zxilly** [comment]
@hsluoyz He is applying for Summer 2021

**Comment by captt-g** [comment]
@captt-g Now run the test.py, I can get message like 
image
,plz review, if there are something wrong, I will fix it.

**Comment by captt-g** [comment]
@techoner Now run the test.py, I can get message like 
4E18169A-85CD-4938-BF4C-573474D0B3E1
,plz review, if there are something wrong, plz tell me, I will fix it.

**Comment by captt-g** [comment]
@hsluoyz Hi,I'm applying for Summer 2021.

**Comment by hsluoyz** [comment]
@captt-g plz contact community as: https://github.com/casbin/Summer2021#%E6%8A%95%E9%80%92%E8%A6%81%E6%B1%82

**Comment by Zxilly** [comment]
@captt-g please remove `.idea` files.

**Comment by hsluoyz** [comment]
@captt-g squash commits

**Comment by ffyuanda** [comment]
@captt-g before claiming your work, do you need to consult the original author of these codes? [https://github.com/ScienceLogic/pycasbin-redis-watcher]()

**Comment by captt-g** [comment]
@ffyuanda I have refer to his code ,but not copy word to word, his code isn't work because long time no vindicate.

**Comment by ffyuanda** [comment]
@captt-g I'm not saying you are plagiarizing, it is all open-sourced. My thought is it will look better if you claim it more obviously and accredit some of the work back to the original repo. Your work is good.

**Comment by captt-g** [comment]
@ffyuanda  I don't think it's necessary.Because I also refer to many other repo ,like https://github.com/casbin/redis-watcher and https://github.com/pycasbin/postgresql-watcher,they are also very useful.But my mind runs through my work.

**Comment by ffyuanda** [comment]
@captt-g good for you then

**Comment by captt-g** [comment]
@ffyuanda Thanks, I have add the repo I have refer to in the file commits now.

**Review by techoner — CHANGES_REQUESTED** [review]

**Comment by hsluoyz** [comment]
@captt-g why close?

**Comment by captt-g** [comment]
I think this work may not fulfil the requirements.

**Pull request closed without merging**